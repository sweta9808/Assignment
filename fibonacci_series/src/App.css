.App {
  text-align: center;
}
.fibonacciItem{ padding-right: 5px;}
.compositeNum{color: green;}
.primeNum{color: blue;}
.pageNumber{padding-right: 5px;cursor: pointer;}