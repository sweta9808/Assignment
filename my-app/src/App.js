import React, { Component } from 'react';

//APP
class App extends Component {
    render() {
      const list = this.props.list && this.props.list.map(id => {
        return <option key={id} value={id}>{id}</option>
      })
  
      const departments = this.props.departments && this.props.departments.map(dept => {
        return <option key={dept} value={dept}>{dept}</option>
      })
  
      const details = <>
        <img src={this.props.data && this.props.data.avatar}/>
        <div><strong>ID: </strong>{this.props.data && this.props.data.id}</div>
        <div>{this.props.data && this.props.data.first_name + " "}{this.props.data && this.props.data.last_name}</div>
      </>
  
      return (<>
          <div className="Input row container">
            <div >
            <label>Department</label>
              <select id="department" data-testid="select-department" onChange={this.props.handleDepartmentChange}>
                {departments}
              </select>
            </div>
            <div>
            <label>Employment ID</label>
              <select id="emp" data-testid="emp-id" onChange={this.props.handleEmployeeIdChange}>
                {list}
              </select>
            </div>
  
            <div>
                <button id="button" data-testid="get-details" onClick={() => this.props.handleShowDetails(`https://reqres.in/api/users/${this.props.employee}`)}>Show Details</button>
            </div>
            <div>
                <button id="clear" data-testid="clear-all" onClick={this.props.handleClear}>Clear</button>
            </div>
          </div>
  
          <div className="Output">
            {this.props.showDetails && details}
          </div>
          </>
      )
    }
  
    componentDidMount() {
      this.props.adjustUI(this.props.departments[0], this.props.list[0])
    }
  }

  export default App;
