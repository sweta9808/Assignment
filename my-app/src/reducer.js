export const initialState = {
    department: null,
    employee: null,
    departments: ["HR", "ENGINEERS"],
    hrs: [1,2,3,4,5],
    engineers: [6,7,8,9],
    list: [1,2,3,4,5],
    data: null,
    showDetails: false
  };

//REDUCER
export const reducer = function(state=initialState, {type, payload}) {
    const newState = {...state}
    switch(type) {
      case "UPDATE_DEPARTMENT_AND_EMPLOYEE":
        return {...state, ...payload}
      case "UPDATE_DEPARTMENT_ASYNC":
        return {...state, ...payload}
      case "UPDATE_EMPLOYEE":
        return {...state, ...payload}
      case "UPDATE_DATA_ASYNC":
        return {...state, ...payload}
      case "CLEAR":
        return {...state, ...payload}
      default: 
        return newState
    }
  }
  