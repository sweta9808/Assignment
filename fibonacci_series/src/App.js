import React from 'react';
import './App.css';

class App extends React.Component {
  constructor(){
    super();
    this.state={
      fibonacciSeries:[],
      isClicked:false,
      currentPage: 1,
      todosPerPage: 10,
      value:''
    }
  }
  handleClick =(event) =>{
    this.setState({
      currentPage: Number(event.target.id)
    });
  }
  fibonacci = (event) => {
    let num = event.target.value;
    if (isNaN(num) || num < 0 || event.target.value[0] === '0' || num > 2000) {
      num = '';
     }
     this.setState({value:num,isClicked:false})
    let fib = [0, 1];
    let data = [];
    
    for(let i = 2; i <= num; i++) {
      fib[i] = fib[i - 1] + fib[i - 2]; 
      data.push(fib[i]);
    }
    let fibonacciSeries = data.filter((item) => {
      if(item<num) {
        return item;
      }
    })
    this.setState({fibonacciSeries:fibonacciSeries})
  }
  
  quicksort = (arr) => {
    if (arr.length <= 1)
    return arr;
  let pp = Math.floor(arr.length / 2), pivot = arr[pp];
  const left = [], right = [];
  for (var i = 0; i < arr.length; i++) {
    if (i == pp) continue;
    if (arr[i] < pivot) {
      left.push(arr[i]);
    }
    else {
      right.push(arr[i]);
    }
  }
  return this.quicksort(left).concat(pivot, this.quicksort(right));

  }
  handleSort =() =>{
   let sortedArray =this.quicksort(this.state.fibonacciSeries);
   this.setState({fibonacciSeries:sortedArray,isClicked:true})
  }
  render(){
    const { fibonacciSeries, currentPage, todosPerPage } = this.state;
    // Logic for displaying current todos
    const indexOfLastTodo = currentPage * todosPerPage;
    const indexOfFirstTodo = indexOfLastTodo - todosPerPage;
    const currentTodos = fibonacciSeries.slice(indexOfFirstTodo, indexOfLastTodo);
    const renderTodos = currentTodos.map((todo, index) => {
      return <span className="fibonacciItem" key={index}>{todo}</span>;
    });

    // Logic for displaying page numbers
    const pageNumbers = [];
    console.log(fibonacciSeries);
    console.log(todosPerPage);
    for (let i = 1; i <= Math.ceil(fibonacciSeries.length / todosPerPage); i++) {
      pageNumbers.push(i);
    }

    const renderPageNumbers = pageNumbers.map(number => {
      return (
        <span className="pageNumber"
          key={number}
          id={number}
          onClick={this.handleClick}
        >
          {number}
        </span>
      );
    });

    return (
    <div className="App">
      <div className="form-group">
        <input type="number"
          className="form-control" name="number" value={this.state.value} id="number" pattern="\d+" min="1" max="2000" onChange={this.fibonacci} aria-describedby="input" />      
      </div>
     
      {this.state.fibonacciSeries?.length>1?<div>
        <button type="button" className="btn btn-primary" data-toggle="button" onClick={this.handleSort} aria-pressed="false" autocomplete="off">Sort</button>
      </div>:''}
      {this.state.isClicked?<div>
        {currentTodos.map((item, index) => {
          for(let i = 2; i < item; i++) {
            if(item % i === 0) return <span className="fibonacciItem compositeNum" key={index}>{item}-Comp </span>;
            return <span className="fibonacciItem primeNum" key={index}>{item}-Prime</span>;
          }
        })}
      </div>:<div> {renderTodos}</div>}
        <div id="page-numbers">
          {renderPageNumbers}
        </div>
    </div>
  );
 }
}

export default App;
