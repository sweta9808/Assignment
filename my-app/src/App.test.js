import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render, fireEvent } from '@testing-library/react';

import App from './App';

import { reducer, initialState } from './reducer';

const configStore = createStore(reducer, initialState);

const renderWithRedux = (
    component,
    { initialState, store = configStore } = {}
  ) => {
    return {
      ...render(<Provider store={store}>{component}</Provider>),
      store,
    }
  }
 
describe('App Test Cases', () => {

    const props = {
        adjustUI: () => {},
        ...initialState,
    }

    it('should take a snapshot', () => {
        const { asFragment } = renderWithRedux(<App {...props} />)
        
        expect(asFragment(<App {...props} />)).toMatchSnapshot()
    });
    
    it('should Department change', () => {
        const { getByTestId } = renderWithRedux(<App {...props} />); 

        fireEvent.change(getByTestId('select-department'));
        fireEvent.change(getByTestId('emp-id'));
        fireEvent.change(getByTestId('get-details'));
        fireEvent.change(getByTestId('clear-all'));
    });

});
