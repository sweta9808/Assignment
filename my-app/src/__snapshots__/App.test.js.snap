// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Cases should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="Input row container"
  >
    <div>
      <label>
        Department
      </label>
      <select
        data-testid="select-department"
        id="department"
      >
        <option
          value="HR"
        >
          HR
        </option>
        <option
          value="ENGINEERS"
        >
          ENGINEERS
        </option>
      </select>
    </div>
    <div>
      <label>
        Employment ID
      </label>
      <select
        data-testid="emp-id"
        id="emp"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
      </select>
    </div>
    <div>
      <button
        data-testid="get-details"
        id="button"
      >
        Show Details
      </button>
    </div>
    <div>
      <button
        data-testid="clear-all"
        id="clear"
      >
        Clear
      </button>
    </div>
  </div>
  <div
    class="Output"
  />
</DocumentFragment>
`;
